File: leetcode/medium/Q11_ContainerWithMostWater.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        answer = 0
        left = 0
        right = len(height) - 1

        while left < right:
            min_h = min(height[left], height[right])
            answer = max(answer, (min_h * (right-left)))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return answer

File: src/medium/Q11_ContainerWithMostWater.java
package medium;

public class Q11_ContainerWithMostWater {
    class Solution {
        public int maxArea(int[] height) {
            int length = height.length;
            int start = 0;
            int end = length -1;
            int max = 0;
            while(start < end){
                max = Math.max(max, Math.min(height[start], height[end]) * (end - start));
                if(height[start] < height[end]) {
                    start += 1;
                }else{
                    end -= 1;
                }
            }
            return max;
        }
    }
}
